### PR TITLE
`purs-tidy` for PureScript

### PR DESCRIPTION
**Pull request by toastal**

### CODE REVIEW

File: test/fixers/test_purs_tidy_fixer_callback.vader [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Before:
```
Comment (hsanson) [inline test/fixers/test_purs_tidy_fixer_callback.vader:1]: Check https://github.com/dense-analysis/ale/blob/master/test/fixers/test_dhall_freeze_fixer_callback.vader for a simpler way to test a fixer using `ale#assert#SetUpFixerTest()`, `ale#assert#TearDownFixerTest()`, and `AssertFixer`.

**Comment by toastal** [comment]
What on earth does this test error mean?

File: autoload/ale/fixers/purs_tidy.vim [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    \   'node_modules/.bin/purs-tidy',
+    \])
+endfunction
+
+function! ale#fixers#purs_tidy#Fix(buffer) abort
+    let l:executable = ale#fixers#purs_tidy#GetExecutable(a:buffer)
+    let l:options = ale#Var(a:buffer, 'purescript_tidy_options')
+
+    return {
+    \   'command': l:executable
```
Comment (hsanson) [inline autoload/ale/fixers/purs_tidy.vim:20]: Escape this:

```
\   'command': ale#Escape(l:executable)
```

File: test/fixers/test_purs_tidy_fixer_callback.vader [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Execute(The purs-tidy callback should return the correct custom options):
+  AssertFixer
+  \ {
+  \   'command': ale#Escape('odd-purs-tidy')
+  \     . ' --indent 3'
+  \     . ' %s',
+  \   'read_temporary_file': 0,
+  \ },
+  \ ale#fixers#purs_tidy#Fix(bufnr(''))
```
Comment (hsanson) [inline test/fixers/test_purs_tidy_fixer_callback.vader:22]: This last line `ale#fixers#purs_tidy#Fix(bufnr(''))` is not needed.

**Review by hsanson — CHANGES_REQUESTED** [review]

File: autoload/ale/fixers/purs_tidy.vim (hunk 2 above)
Comment (toastal) [reply autoload/ale/fixers/purs_tidy.vim]: thanks!

File: test/fixers/test_purs_tidy_fixer_callback.vader (hunk 3 above)
Comment (toastal) [reply test/fixers/test_purs_tidy_fixer_callback.vader]: thanks!

**Comment by toastal** [comment]
Thank you @hsanson . It looks like we are all green now.

**Review by hsanson — APPROVED** [review]
Thanks for the contribution.

**Pull request merged**